File: framework/core/src/main/java/cn/wjybxx/bigcat/fx/WorkerEvent.java
/*
 * Copyright 2023-2024 wjybxx(845740757@qq.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package cn.wjybxx.bigcat.fx;

import cn.wjybxx.concurrent.IAgentEvent;

/**
 * Worker主循环事件
 * (是池化的，不必担心开销)
 *
 * @author wjybxx
 * date - 2025/3/14
 */
public final class WorkerEvent implements IAgentEvent {

    private int type = TYPE_INVALID;
    private int options;
    public Object obj1;
    public Object obj2;
    public Object obj3;

    // 扩展字段
    public int intVal1;
    public int intVal2;
    public long longVal1;
    public long longVal2;

    @Override
    public void clean() {
        type = TYPE_INVALID;
        options = 0;
        obj1 = null;
        obj2 = null;
        obj3 = null;
    }

    @Override
    public void cleanAll() {
        type = TYPE_INVALID;
        options = 0;
        obj1 = null;
        obj2 = null;
        obj3 = null;

        intVal1 = 0;
        intVal2 = 0;
        longVal1 = 0;
        longVal2 = 0;
    }

    @Override
    public int getType() {
        return type;
    }

    @Override
    public void setType(int type) {
        this.type = type;
    }

    @Override
    public int getOptions() {
        return options;
    }

    @Override
    public void setOptions(int options) {
        this.options = options;
    }

    @Override
    public Object getObj1() {
        return obj1;
    }

    @Override
    public void setObj1(Object obj1) {
        this.obj1 = obj1;
    }

    @Override
    public Object getObj2() {
        return obj2;
    }

    @Override
    public void setObj2(Object obj2) {
        this.obj2 = obj2;
    }

    public Object getObj3() {
        return obj3;
    }

    public void setObj3(Object obj3) {
        this.obj3 = obj3;
    }

    public int getIntVal1() {
        return intVal1;
    }

    public void setIntVal1(int intVal1) {
        this.intVal1 = intVal1;
    }

    public int getIntVal2() {
        return intVal2;
    }

    public void setIntVal2(int intVal2) {
        this.intVal2 = intVal2;
    }

    @Override
    public long getLongVal1() {
        return longVal1;
    }

    @Override
    public void setLongVal1(long longVal1) {
        this.longVal1 = longVal1;
    }

    @Override
    public long getLongVal2() {
        return longVal2;
    }

    @Override
    public void setLongVal2(long longVal2) {
        this.longVal2 = longVal2;
    }
}

File: csharp/Wjybxx.BigCat.Core/src/Fx/WorkerEvent.cs
#region LICENSE

// Copyright 2025 wjybxx(845740757@qq.com)
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

#endregion

using System;
using Wjybxx.Commons.Concurrent;

namespace Wjybxx.BigCat.Fx
{
/// <summary>
/// Worker线程上支持的事件
/// (以后再删除不必要的字段，正式项目会用完的)
/// </summary>
public struct WorkerEvent : IAgentEvent
{
#nullable disable
    private int type;
    private int options;
    public int intVal1;
    public int intVal2;
    public long longVal1;
    public long longVal2;
    public object obj1;
    public object obj2;
    public object obj3;

    /// <summary>
    /// 构造函数将type声明为可选值，会导致不被调用构造函数
    /// </summary>
    public static readonly Func<WorkerEvent> FACTORY = () => {
        WorkerEvent r = default;
        r.type = IAgentEvent.TYPE_INVALID;
        return r;
    };

    public WorkerEvent(int type) : this() {
        this.type = type;
    }

    public int Type {
        get => type;
        set => type = value;
    }

    public int Options {
        get => options;
        set => options = value;
    }

    public object Obj1 {
        get => obj1;
        set => obj1 = value;
    }

    public object Obj2 {
        get => obj2;
        set => obj2 = value;
    }

    public object Obj3 {
        get => obj3;
        set => obj3 = value;
    }

    public long LongVal1 {
        get => longVal1;
        set => longVal1 = value;
    }
    public long LongVal2 {
        get => longVal2;
        set => longVal2 = value;
    }

    public int IntVal1 {
        get => intVal1;
        set => intVal1 = value;
    }
    public int IntVal2 {
        get => intVal2;
        set => intVal2 = value;
    }

    public void Clean() {
        type = IAgentEvent.TYPE_INVALID;
        options = 0;
        obj1 = null;
        obj2 = null;
        obj3 = null;
    }

    public void CleanAll() {
        type = IAgentEvent.TYPE_INVALID;
        options = 0;
        obj1 = null;
        obj2 = null;
        obj3 = null;
        intVal1 = 0;
        intVal2 = 0;
        longVal1 = 0;
        longVal2 = 0;
    }

    public override string ToString() {
        return $"{nameof(type)}: {type}," +
               $" {nameof(options)}: {options}," +
               $" {nameof(obj1)}: {obj1}," +
               $" {nameof(obj2)}: {obj2}," +
               $" {nameof(obj3)}: {obj3}," +
               $" {nameof(intVal1)}: {intVal1}," +
               $" {nameof(intVal2)}: {intVal2}," +
               $" {nameof(longVal1)}: {longVal1}," +
               $" {nameof(longVal2)}: {longVal2}";
    }
}
}